src/OpenFeature/OpenFeatureClient.cs

```diff
@@ -262,7 +262,7 @@ public void AddHooks(IEnumerable<Hook> hooks)
             {
                 this._logger.LogError(ex, "Error while evaluating flag {FlagKey}", flagKey);
                 var errorCode = ex is InvalidCastException ? ErrorType.TypeMismatch : ErrorType.General;

```

Comment:
It looks like this line helps find how we could test this.

There is a test `OpenFeatureClient_Should_Return_DefaultValue_When_Type_Mismatch` which triggers this condition, and now it will have a message. 

```
            var evaluationDetails = await client.GetObjectDetails(flagName, defaultValue);
            evaluationDetails.ErrorMessage.Should().Be(new InvalidCastException().Message);
```

Reply:
Oh, thanks it works 🙌 

Coverage is still not working, but it is not related to anything introduced by this PR.